File: app/src/test/java/com/example/androidfundamentals/di/MockAddressComponent.java
package com.example.androidfundamentals.di;

import android.app.Application;
import android.content.Context;

import com.example.androidfundamentals.activities.AddressActivity;
import com.example.androidfundamentals.activities.AddressActivityTest;

import dagger.Component;

@Component(modules = {MockAddressModule.class})
public interface MockAddressComponent {
    void inject(AddressActivityTest activityTest);
}

File: app/src/test/java/com/example/androidfundamentals/di/MockAddressComponent.kt
package com.example.androidfundamentals.di

import android.content.Context
import com.example.androidfundamentals.activities.AddressActivityTest
import dagger.BindsInstance

import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Component(modules = [AndroidInjectionModule::class, MockAddressModule::class, ApplicationActivitiesModule::class])
@Singleton
interface MockApplciationAndroidComponent: ApplicationAndroidComponent {

    @Component.Factory
    interface Factory {
        fun create(@BindsInstance applicationContext: Context): ApplicationAndroidComponent
    }

    fun inject(addressActivity: AddressActivityTest)
}
